tests/Unit/Evolution/Systems/ScalarTensor/Test_TimeDerivativeTerms.cpp

```diff
@@ -0,0 +1,259 @@
+      const gsl::not_null<EvolvedVariables*> dt_vars_ptr,
+      [[maybe_unused]] const gsl::not_null<FluxVariables*> volume_fluxes,
+      const gsl::not_null<TemporaryVariables*> temporaries,
+      const GradientVariables& partial_derivs,
+      const ArgumentVariables& time_derivative_args) {
+    ComputeVolumeTimeDerivativeTerms::apply(
+        make_not_null(&get<::Tags::dt<EvolvedTags>>(*dt_vars_ptr))...,
+        make_not_null(&get<FluxTags>(*volume_fluxes))...,
+        make_not_null(&get<TempTags>(*temporaries))...,
+        get<GradientTags>(partial_derivs)..., [](const auto& t) -> const auto& {
+          if constexpr (tt::is_a_v<std::unique_ptr,
+                                   std::decay_t<decltype(t)>>) {
+            return *t;
+          } else {
+            return t;
+          }
+        }(tuples::get<ArgTags>(time_derivative_args))...);
+  }
+
+  template <typename EvolvedVariables, typename FluxVariables,
+            typename TemporaryVariables, typename GradientVariables,
+            typename ArgumentVariables>
+  static void apply_packed(
+      const gsl::not_null<EvolvedVariables*> dt_vars_ptr,
+      [[maybe_unused]] const gsl::not_null<FluxVariables*> volume_fluxes,
+      const gsl::not_null<TemporaryVariables*> temporaries,
+      const GradientVariables& partial_derivs,
+      const ArgumentVariables& time_derivative_args) {
+    ComputeVolumeTimeDerivativeTerms::apply(
+        dt_vars_ptr, volume_fluxes, temporaries,
+        get<GradientTags>(partial_derivs)..., [](const auto& t) -> const auto& {
+          if constexpr (tt::is_a_v<std::unique_ptr,
+                                   std::decay_t<decltype(t)>>) {
+            return *t;
+          } else {
+            return t;
+          }
+        }(tuples::get<ArgTags>(time_derivative_args))...);
+  }
+};
+
+} // namespace
+
+SPECTRE_TEST_CASE(
+    "Unit.Evolution.Systems.ScalarTensor.TimeDerivativeTerms",
+    "[Unit][Evolution]") {
+  using gh_variables_tags =
+      typename gh::System<3_st>::variables_tag::tags_list;
+  using scalar_variables_tags =
+      typename CurvedScalarWave::System<3_st>::variables_tag::tags_list;
+
+  using gh_dt_variables_tags =
+      ScalarTensor::TimeDerivativeTerms::gh_dt_tags;
+  using scalar_dt_variables_tags =
+      ScalarTensor::TimeDerivativeTerms::scalar_dt_tags;
+  using dt_variables_type = Variables<
+      tmpl::append<gh_dt_variables_tags, scalar_dt_variables_tags>>;
+
+  using gh_flux_tags = tmpl::list<>;
+  using scalar_flux_tags =
+      ScalarTensor::TimeDerivativeTerms::scalar_flux_tags;
+
+  using flux_variables_type =
+      Variables<tmpl::append<gh_flux_tags, scalar_flux_tags>>;
+
+  using gh_temp_tags = ScalarTensor::TimeDerivativeTerms::gh_temp_tags;
+  using scalar_temp_tags =
+      ScalarTensor::TimeDerivativeTerms::scalar_temp_tags;
+  using temp_variables_type = Variables<
+      typename ScalarTensor::TimeDerivativeTerms::temporary_tags>;
+
+  using gh_gradient_tags = tmpl::transform<
```

Comment:
doesn't `using gh_gradient_tags = ScalarTensor::TimeDerivativeTerms::gh_gradient_tags` work fine?

Reply:
Tried it, doesn't seem to work for the Helper

Reply:
ah right, I misread this. The gradient tags in the system don't actually have the derivative tag attached to it yet and this metafunction does that. fine as is